
<launch>

  <arg name="world_name" value="$(find hexapod)/worlds/pigot.world"/>
  <arg name="paused" default="true"/>
  <arg name="use_sim_time" default="true"/>
  <arg name="gui" default="true"/>
  <arg name="headless" default="false"/>
  <arg name="debug" default="false"/>

  <!-- We resume the logic in empty_world.launch, changing only the name of the world to be launched -->
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find hexapod)/worlds/pigot.world"/>
    <arg name="debug" value="$(arg debug)" />
    <arg name="gui" value="$(arg gui)" />
    <arg name="paused" value="$(arg paused)"/>
    <arg name="use_sim_time" value="$(arg use_sim_time)"/>
    <arg name="headless" value="$(arg headless)"/>
  </include>


    <param name="robot_description"
    command="$(find xacro)/xacro --inorder '$(find hexapod)/urdf/hexapod.xacro'" />

    <!-- 在 gazebo 中显示机器人模型 -->
    <node pkg="gazebo_ros" type="spawn_model" name="model" args="-urdf -model hexapod -param robot_description -x 0 -y 0 -z 0.2"  />


    <include file="$(find hexapod_pkg)/launch/control.launch" />

    <node name="pos_pub_node" pkg="hexapod_pkg" type="pub.py" respawn="true"/>


</launch>